client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -402,6 +416,105 @@ private void checkIfBlocksFailed() {
     }
   }
 
+  private boolean needReAssignShuffleServer(Set<Long> failedBlockIds) {
+    boolean result = true;
+    FailedBlockSendTracker failedBlockTracker = shuffleManager.getBlockIdsFailedSendTracker(taskId);
```

Comment:
`failedBlockTracker` will be null, I don't see any initialization process about this

Reply:
do it. and use `FailedBlockSendTracker failedBlockSendTracker ` instead of  
```
  Map<Long, List<ShuffleServerInfo>> sendFailedBlockIds;
  Map<Long, BlockingQueue<ShuffleServerInfo>> sendFailedBlockIds;

```